cpp/simulations/abm.cpp

```diff
@@ -1102,33 +670,35 @@ mio::IOResult<void> run(const fs::path& result_dir, size_t num_runs, bool save_s
     auto tmax             = mio::abm::TimePoint(0) + mio::abm::days(60); // End time per simulation
     auto ensemble_results = std::vector<std::vector<mio::TimeSeries<ScalarType>>>{}; // Vector of collected results
     ensemble_results.reserve(size_t(num_runs));
+    auto ensemble_params = std::vector<std::vector<mio::abm::World>>{};
+    ensemble_params.reserve(size_t(num_runs));
     auto run_idx            = size_t(1); // The run index
     auto save_result_result = mio::IOResult<void>(mio::success()); // Variable informing over successful IO operations
-
+    std::vector<int> loc_ids;
     // Loop over a number of runs
     while (run_idx <= num_runs) {
-
-        // Create the sampled simulation with start time t0.
+        loc_ids = {};
+        // Create the sampled simulation with start time t0
         auto sim = create_sampled_simulation(t0);
```

Comment:
Here, I would suggest to use the copy operation of the world so that we don't need to create everything from data/scratch all over again. This might be fine in this context, but if we generate from data this is very expensive for each run. Rather create one world before the loop and then copy from it all the time when creating a simulation for a run.
This will, however, remove the randomness in initialisation, which we should discuss.

Reply:
Yes, I moved the initialisation of the World to be outside the loop.